presto-main/src/main/java/io/prestosql/sql/planner/NodePartitioningManager.java

```diff
@@ -203,7 +203,7 @@ private ConnectorBucketNodeMap getConnectorBucketNodeMap(Session session, Partit
 
         ConnectorBucketNodeMap connectorBucketNodeMap = partitioningProvider.getBucketNodeMap(
                 partitioningHandle.getTransactionHandle().orElse(null),
-                session.toConnectorSession(),
+                session.toConnectorSession(partitioningHandle.getConnectorId().get()),
```

Comment:
Passing catalog name to `toConnectorSession` is about passing proper information about identity like what authorization roles are set for identity for given catalog.

The best would be to simply deprecate `toConnectorSession()` (no-arg), but sometimes we do not operate that for any specific connector or simply we don't care about authorization roles. 

Typically authorization roles are used only in access control.

Why you changes these methods and left many other `toConnectorSession()` untouched. Are you have custom plugin or do I miss something?